flask-api/app.py

```diff
@@ -40,12 +41,165 @@ def serialize(self):
+
+class PatientByIdResource(Resource):
+    def get(self, patient_id):
+        patient = Patient.query.get(patient_id)
+        if patient:
+            return patient.serialize()
+        else:
+            return {'message': 'Patient not found'}, 404
+
+    def put(self, patient_id):
+        patient = Patient.query.get(patient_id)
+        if patient:
+            json_data = request.get_json()
+
+            patient.name = json_data.get('name')
+            patient.day_of_birth = json_data.get('day_of_birth')
+            patient.month_of_birth = json_data.get('month_of_birth')
+            patient.year_of_birth = json_data.get('year_of_birth')
+            patient.sex = json_data.get('sex')
+            patient.phone = json_data.get('phone')
+            patient.village = json_data.get('village')
+            patient.natID = json_data.get('natID')
+            patient.photo_thumbnail = json_data.get('photo_thumbnail')
+            patient.voice_clip = json_data.get('voice_clip')
+            patient.photo_file_path = json_data.get('photo_file_path')
+            patient.voice_recording_path = json_data.get('voice_recording_path')
+
+            db.session.commit()
+            return {'message': 'Patient record updated successfully'}
+        else:
+            return {'message': 'Patient not found'}, 404
+
+class ClinicNoteResources(Resource):
+    def get(self):
+        clinic_notes = ClinicNote.query.filter_by(patient_id=json_data.get('patient_id')).all()
+        return [clinic_note.serialize() for clinic_note in clinic_notes]
+
+    def post(self):
+        json_data = request.get_json()
+
+        clinic_note = ClinicNote(
+            patient_id=json_data.get('patient_id'),
+            title=json_data.get('title'),
+            date=json_data.get('date'),
+            notes=json_data.get('notes')
+        )
+
+        db.session.add(clinic_note)
+        db.session.commit()
+
+        return {'message': 'Clinic note created successfully'}, 201
+
+class ClinicNoteByIdResource(Resource):
+    def get(self, clinic_note_id):
+        clinic_note = ClinicNote.query.get(clinic_note_id)
+        if clinic_note:
+            return clinic_note.serialize()
+        else:
+            return {'message': 'Clinic note not found'}, 404
+
+    def put(self, clinic_note_id):
+        clinic_note = ClinicNote.query.get(clinic_note_id)
+        if clinic_note:
+            json_data = request.get_json()
+
+            clinic_note.title = json_data.get('title')
+            clinic_note.date = json_data.get('date')
+            clinic_note.notes = json_data.get('notes')
+
+            db.session.commit()
+            return {'message': 'Clinic note updated successfully'}
+        else:
+            return {'message': 'Clinic note not found'}, 404
+
+class PatientOverviewByIdResource(Resource):
+    def get(self, patient_id):
+        patient_overview = PatientOverview.query.filter_by(patient_id=patient_id).first()
```

Comment:
not sure if this is the right way to filter using the foreign key?

Also this PatientOverview doesn't technically need to be a different table since there is one PatientOverview for each Patient. I separated it like this since not all the pages need this overview just one. I also didn't want to drop the table you made lol